cockpit/src/insights.jsx

```diff
@@ -0,0 +1,292 @@
+}
+
+function get_lastupload_timestamp() {
+    return cockpit.spawn([ "stat", "-c", "%Y", "/etc/insights-client/.lastupload" ]).then(parseInt);
+}
+
+function show_status_dialog() {
+    function show(lastupload, monotonic_start) {
+        let next_elapse = calc_next_elapse(monotonic_start, insights_timer.details);
+
+        let failed_text = _("A recent Insights data upload has failed.");
+        if (insights_service.unit.ActiveExitTimestamp &&
+            insights_service.unit.ActiveExitTimestamp / 1e6 > lastupload) {
+            lastupload = insights_service.unit.ActiveExitTimestamp / 1e6;
+            failed_text = _("The last Insights data upload has failed.");
+        }
+
+        let dlg = show_modal_dialog(
+            {
+                title: _("Connected to Red Hat Insights"),
+                body: (
+                        <div className="modal-body">
+                          <table>
+                            <tbody>
+                              <tr>
+                                <th style={{ textAlign: "right", paddingRight: "1em" }}>Next Insights data upload</th>
+                                <td>{next_elapse}</td>
+                              </tr>
+                              { lastupload &&
+                              <tr>
+                                <th style={{ textAlign: "right", paddingRight: "1em" }}>Last Insights data upload</th>
+                                <td>{moment(lastupload * 1000).calendar()}</td>
+                              </tr>
+                              }
+                            </tbody>
+                          </table>
+                          <br/>
+                          { insights_timer.state == "failed" &&
+                          <div className="alert alert-warning">
+                            <span className="pficon pficon-warning-triangle-o"/>
+                              {_("Next Insights data upload could not be scheduled.")}{" "}
+                              <a onClick={left(jump_to_timer)}>{_("Details")}</a>
+                          </div>
+                          }
+                          { insights_service.state == "failed" &&
+                          <div className="alert alert-warning">
+                            <span className="pficon pficon-warning-triangle-o"/>
+                              {failed_text}{" "}
+                              <a onClick={left(jump_to_service)}>{_("Details")}</a>
+                          </div>
+                          }
+                          <Revealer summary={_("Disconnect from Insights")}>
+                            <div className="alert alert-warning"
+                                 style={{ "padding": "14px", "marginTop": "1ex", "marginBottom": "0px" }}>
+                              <p>{_("If you disconnect this system from Insights, it will no longer report it's Insights status in Customer Portal or Satellite.")}</p>
```

Comment:
I think this needs to say **Red Hat Cloud** rather than **Customer Portal** now.

Reply:
Done.